CRAMv3.tex

```diff
@@ -1539,31 +1542,80 @@ \subsubsection*{General notes}
 must always be read before decoding a slice. Therefore two read operations are 
 required for random access in CRAM.
 
-Indexing a CRAM file is deemed to be a lightweight operation because it does not 
-require any CRAM records to be read. All indexing information can be obtained from 
-container headers, namely sequence id, alignment start and span, container start 
-byte offset and slice byte offset inside the container. 
+Indexing a CRAM file is deemed to be a lightweight operation because it usually does not require any CRAM records to be read.
+Indexing information can be obtained from container headers, namely sequence id, alignment start and span, container start byte offset and slice byte offset inside the container (landmarks).
+The exception to this is with multi-reference containers, where the ``RI'' data series must be read.
 
 \subsubsection*{CRAM index}
 
 A CRAM index is a gzipped tab delimited file containing the following columns:
 
 \begin{enumerate}
-\item Sequence id
+\item Reference sequence id
 
 \item Alignment start (ignored on read for unmapped slices, set to 0 on write)
 
 \item Alignment span (ignored on read for unmapped slices, set to 0 on write)
 
-\item Container start byte offset in the file
+\item Absolute byte offset of Container structure in the file.
 
-\item Slice start byte offset in the container data (`blocks')
+\item Relative byte offset of the Slice header block, from the end of
+the container structure.  This is the same as the ``landmark'' field
+in the container structure.
 
-\item Slice bytes
+\item Slice size in bytes (including slice header and all blocks).
 \end{enumerate}
 
-Each line represents a slice in the CRAM file. Please note that all slices must 
-be listed in the index file.
+Each line represents a slice in the CRAM file.
+Please note that all slices must be listed in the index file.
+
+Multi-reference slices may need to have multiple lines for the same slice; one for each reference contained within that slice.
+In this case the index reference sequence ID will be the actual reference number (from the ``RI'' data series) and not -2.
+
+Slices containing solely unmapped unplaced data (reference ID -1) still require values for all columns, although the alignment start and span will be ignored.
+It is recommended that they are set to small constant values (0 or 1).
```

Comment:
Rather embarrassingly I put 0 or 1 because my own code uses both. ;-)

It doesn't actually matter as we define them to be ignored on read, but a constant single byte value keeps the index small and compressible.  I'll just change it to zero I think and at some point fix my code.